packages/editor/src/store/selectors.js

```diff
@@ -728,16 +727,17 @@ export const getClientIdsWithDescendants = createSelector(
  */
 export const getGlobalBlockCount = createSelector(
 	( state, blockName ) => {
+		const clientIds = getClientIdsWithDescendants( state );
 		if ( ! blockName ) {
-			return size( state.editor.present.blocks.byClientId );
+			return clientIds.length;
 		}
-		return reduce(
-			state.editor.present.blocks.byClientId,
-			( count, block ) => block.name === blockName ? count + 1 : count,
-			0
-		);
+		return reduce( clientIds, ( count, clientId ) => {
+			const block = state.editor.present.blocks.byClientId[ clientId ];
+			return block.name === blockName ? count + 1 : count;
+		}, 0 );
 	},
 	( state ) => [
+		state.editor.present.blocks.order,
 		state.editor.present.blocks.byClientId,
```

Comment:
Is it really worth memoizing selectors if they depend on these two elements particularly? I feel `state.editor.present.blocks.byClientId` changes so often that it's just a waste of time to memoize these selectors.

Reply:
Thoughts on this @aduth?

Reply:
But it seems wasteful to have this selector run when something unrelated e.g. a post attribute is modified, no?

Reply:
I don't know honestly, it feels like typing in blocks is the most impactful thing in terms of performance, changing an attribute only get triggered once and not extensively. 

Anyway, just something to keep in mind for now, there's no clear "winner"